File: CartyObjC/CartyObjC/OpenViewController.m
//
//  OpenViewController.m
//  CartyObjC
//
//  Created by GZTD-03-01959 on 2026/1/5.
//

#import "OpenViewController.h"
#import <CartySDK/CartySDK.h>

@interface OpenViewController ()<CTAppOpenAdDelegate>
@property (nonatomic,weak)IBOutlet UILabel *infoLabel;
@property (nonatomic,strong)CTAppOpenAd *openAd;
@end

@implementation OpenViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.openAd = [[CTAppOpenAd alloc] init];
    self.openAd.placementid = @"281217152630";
    self.openAd.delegate = self;
}

-(IBAction)loadAd:(id)sender
{
    self.infoLabel.text = @"Start load Ad";
    [self.openAd loadAd];
}

-(IBAction)showAd:(id)sender
{
    if(self.openAd.isReady)
    {
        self.infoLabel.text = @"Start show Ad";
        [self.openAd showAd:self];
    }
    else
    {
        self.infoLabel.text = @"Ad not Ready";
    }
}

- (void)CTOpenAdDidLoad:(nonnull CTAppOpenAd *)ad {
    self.infoLabel.text = @"Ad loaded";
}

- (void)CTOpenAdDidShow:(nonnull CTAppOpenAd *)ad {
    self.infoLabel.text= @"Ad did show";
}

- (void)CTOpenAdLoadFail:(nonnull CTAppOpenAd *)ad withError:(nonnull NSError *)error {
    self.infoLabel.text = [NSString stringWithFormat:@"Ad load fail %@",error];
}

- (void)CTOpenAdShowFail:(nonnull CTAppOpenAd *)ad withError:(nonnull NSError *)error {
    self.infoLabel.text = [NSString stringWithFormat:@"Ad show fail %@",error];
}

- (void)CTOpenAdDidClick:(nonnull CTAppOpenAd *)ad {
    self.infoLabel.text = @"Ad click";
}

- (void)CTOpenAdDidDismiss:(nonnull CTAppOpenAd *)ad {
    self.infoLabel.text = @"Ad close";
}

@end

File: CartySwift/CartySwift/OpenViewController.swift
//
//  OpenViewController.swift
//  CartySwift
//
//  Created by GZTD-03-01959 on 2026/1/5.
//

import UIKit
import CartySDK

class OpenViewController: UIViewController, CTAppOpenAdDelegate
{
    @IBOutlet var infoLabel:UILabel?
    let openAd = CTAppOpenAd()
    
    override func viewDidLoad()
    {
        super.viewDidLoad()
        openAd.placementid = "281217152630"
        openAd.delegate = self
    }
    
    @IBAction func loadAd()
    {
        infoLabel?.text = "Start load Ad"
        openAd.load();
    }
    
    @IBAction func showAd()
    {
        if(openAd.isReady())
        {
            infoLabel?.text = "Start show Ad"
            openAd.show(self)
        }
        else
        {
            infoLabel?.text = "Ad not Ready"
        }
    }
    
    func ctOpenAdDidLoad(_ ad: CTAppOpenAd) {
        infoLabel?.text = "Ad loaded"
    }
    
    func ctOpenAdLoadFail(_ ad: CTAppOpenAd, withError error: any Error) {
        infoLabel?.text = "Ad load fail \(error)"
    }
    
    func ctOpenAdDidShow(_ ad: CTAppOpenAd) {
        infoLabel?.text = "Ad did show"
    }
    
    func ctOpenAdShowFail(_ ad: CTAppOpenAd, withError error: any Error) {
        infoLabel?.text = "Ad show fail \(error)"
    }
    
    func ctOpenAdDidClick(_ ad: CTAppOpenAd) {
        infoLabel?.text = "Ad click"
    }
    
    func ctOpenAdDidDismiss(_ ad: CTAppOpenAd) {
        infoLabel?.text = "Ad close"
    }
}
